v2/src/components/question/index.tsx

```diff
@@ -1,86 +1,75 @@
-import React, { Children, PropsWithChildren, useState } from "react";
+import React, { Children, PropsWithChildren, useState, useRef, useEffect } from "react";
+
+import "./question.css";
 
 export function Question(props: PropsWithChildren<{
     question: string | (() => JSX.Element)
 }>) {
+    const answerBoxRef = useRef<HTMLDivElement>(null);
 
+    const [extraPaddingToAnswerBox, setExtraPaddingToAnswerBox] = useState(false);
     const [selectedAnsTitle, setSelectedAnsTitle] = useState(undefined);
 
     let resubmitInfoClicked = (event: any) => {
         event.preventDefault();
         setSelectedAnsTitle(undefined);
-    }
+    };
+
+    useEffect(() => {
+        updateAnswerBoxPaddingIfAnswersOverflow();
+    }, [selectedAnsTitle]);
+
+    const updateAnswerBoxPaddingIfAnswersOverflow = () => {
+        if (answerBoxRef !== null && answerBoxRef.current !== null) {
+            const answersInsideContainer = answerBoxRef.current.getElementsByClassName("question-box-answer");
+
+            const messageWidth = 227;
+            const answerBoxWidth = answerBoxRef.current.getBoundingClientRect().width;
+            const availableWidth = answerBoxWidth - messageWidth - 36;
+
+            let totalAnswersWidth = 0;
+            for (const answer of answersInsideContainer) {
+                const width = answer.getBoundingClientRect().width;
+                totalAnswersWidth = totalAnswersWidth + width + 24;
+            }
+            setExtraPaddingToAnswerBox(totalAnswersWidth > availableWidth);
+        }
+    };
```

Comment:
Why was this required? I thought this was just a design change

Reply:
it was needed to be conditional, if the answers overlap the message, then only we add extra padding, otherwise we do not add extra padding

Reply:
What does overlap mean here?

Reply:
Also `const messageWidth = 227;` where did you get this number from? and how will this work on mobile?

Reply:
on iPad Pro, the message `Refresh the page to undo your selection` overlaps the answer buttons.

I got the `const messageWidth = 227;` from the width of `Refresh the page to undo your selection`. Though I realise now it would be better to get the width on execution, otherwise we would need to update this value any time we change the style or content of that message. 

![image](https://user-images.githubusercontent.com/92283713/160792598-603e95bf-c05c-497d-975e-d48227b1e5c8.png)

Reply:
Wouldnt it make more sense to give the text a fixed width based on its parent width instead? (and the same for the buttons container too)

Reply:
Also this happens if I open the preview on an actual iPad Pro

![IMG_64B76BD3DEF6-1](https://user-images.githubusercontent.com/18233774/160796606-bb6006f0-14d8-474d-b698-cb9448d4f83b.jpeg)